opa/opa.go

```diff
@@ -0,0 +1,155 @@
+// Package opa provides client make requests to the Open Policy Agent API
+package opa
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"go.uber.org/zap"
+)
+
+// Client OPA client
+type Client struct {
+	logger *zap.Logger
+	Host   string
+}
+
+// EvalutePolicyRequest OPA evalute policy request
+type EvalutePolicyRequest struct {
+	Input json.RawMessage `json:"input"`
+}
+
+// EvaluatePolicyResponse OPA evaluate policy response
+type EvaluatePolicyResponse struct {
+	Result *EvaluatePolicyResult `json:"result"`
+}
+
+// EvaluatePolicyResult OPA evaluate policy result
+type EvaluatePolicyResult struct {
+	Pass       bool `json:"pass"`
+	Violations []*EvaluatePolicyViolation
+}
+
+// EvaluatePolicyViolation OPA evaulate policy violation
+type EvaluatePolicyViolation struct {
+	Message string `json:"message"`
+}
+
+// PolicyViolation Rego rule conditions
+type PolicyViolation struct {
+	Conditions []byte
+}
+
+// Write Rego rule to IO writer
+func (v *PolicyViolation) Write(w io.Writer) {
+	w.Write([]byte("violation[] {\n"))
+	w.Write(v.Conditions)
+	w.Write([]byte("\n}\n\n"))
+}
+
+// NewClient OpaClient constructor
+func NewClient(logger *zap.Logger, host string) *Client {
+	client := &Client{
+		logger: logger,
+		Host:   host,
+	}
+	return client
+}
+
+// InitializePolicy initializes OPA policy if it does not already exist
+func (opa *Client) InitializePolicy(policy string) ClientError {
```

Comment:
can we add some whitespace between some of the lines of these functions to make it easier to read?